FirebaseRemoteConfig/Sources/FIRRemoteConfig.m

```diff
@@ -619,4 +624,60 @@ - (FIRConfigUpdateListenerRegistration *)addOnConfigUpdateListener:
   return [self->_configRealtime addConfigUpdateListener:listener];
 }
 
+#pragma mark - Rollout
+
+- (void)addRemoteConfigInteropSubscriber:(id<FIRRolloutsStateSubscriber>)subscriber {
+  [[NSNotificationCenter defaultCenter]
+      addObserverForName:RolloutsStateDidChangeNotificationName
+                  object:self
+                   queue:nil
+              usingBlock:^(NSNotification *_Nonnull notification) {
+                FIRRolloutsState *rolloutsState =
+                    notification.userInfo[RolloutsStateDidChangeNotificationName];
+                [subscriber rolloutsStateDidChange:rolloutsState];
+              }];
+  // Pass active rollout metadata stored in persistence if exists while app launched.
+  NSArray<NSDictionary *> *activeRolloutMetadata = self->_configContent.activateRolloutMetadata;
+  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
+    [self notifyRolloutsStateChange:activeRolloutMetadata
+                      versionNumber:self->_settings.lastActiveTemplateVersion];
+  });
+}
+
+- (void)notifyRolloutsStateChange:(NSArray<NSDictionary *> *)rolloutMetadata
+                    versionNumber:(NSString *)versionNumber {
+  if (rolloutMetadata && rolloutMetadata.count > 0 && [versionNumber longLongValue] > 0) {
+    NSMutableArray<FIRRolloutAssignment *> *rolloutsAssignments = [[NSMutableArray alloc] init];
+    for (NSDictionary *metadata in rolloutMetadata) {
+      NSString *rolloutId = metadata[RCNFetchResponseKeyRolloutID];
+      NSString *variantID = metadata[RCNFetchResponseKeyVariantID];
+      NSArray<NSString *> *affectedParameterKeys =
+          metadata[RCNFetchResponseKeyAffectedParameterKeys];
+      if (rolloutId && variantID && affectedParameterKeys) {
+        for (NSString *key in affectedParameterKeys) {
+          FIRRemoteConfigValue *value = [self configValueForKey:key];
+          if (value) {
+            FIRRolloutAssignment *assignment =
+                [[FIRRolloutAssignment alloc] initWithRolloutId:rolloutId
+                                                      variantId:variantID
+                                                templateVersion:[versionNumber longLongValue]
+                                                   parameterKey:key
+                                                 parameterValue:value.stringValue];
+            [rolloutsAssignments addObject:assignment];
+          }
+        }
+      }
+    }
+    if (rolloutsAssignments.count > 0) {
```

Comment:
Is it possible during the app life cycle, user clean all the rollouts and the count becomes 0, in this case we still want to pass the information to Crash?

Reply:
Good catch! Yes, we also need to update the fetch & active metadata to an empty list if all rollout removed.